.pnpmfile.cjs

```
@@ -0,0 +1,69 @@
+// https://pnpm.io/pnpmfile
+
+// This implementation is based on https://gist.github.com/dvins/33b8fb52480149d37cdeb98890244c5b.
+// Changes:
+//  - Added support for complete dependency removal (specify null for newVersion)
+//  - Specify remapPeerDependencies for local needs
+
+//@ts-check
+
+// https://pnpm.io/pnpmfile
+// https://github.com/pnpm/pnpm/issues/4214
+// https://github.com/pnpm/pnpm/issues/5391
+
+const rootPkg = require("./package.json");
+
+console.log(`Checking for package peerDependency overrides`);
+
+const remapPeerDependencies = [
+	// @fluidframework/azure-client 1.x declares a peerDependency on fluid-framework but does not require it.
+	// It should have been an optional peer dependency. We just remove it.
+	{
+		package: "@fluidframework/azure-client",
+		packageVersion: "1.",
+		peerDependency: "fluid-framework",
+		newVersion: null,
+	},
+];
+
+function overridesPeerDependencies(pkg) {
+	if (pkg.peerDependencies) {
+		remapPeerDependencies.map((dep) => {
+			if (pkg.name === dep.package && pkg.version.startsWith(dep.packageVersion)) {
+				console.log(`  - Checking ${pkg.name}@${pkg.version}`); // , pkg.peerDependencies);
+
+				if (dep.peerDependency in pkg.peerDependencies) {
+					try {
+						console.log(
+							`    - Overriding ${pkg.name}@${pkg.version} peerDependency ${dep.peerDependency}@${pkg.peerDependencies[dep.peerDependency]}`,
+						);
+
+						// First add a new dependency to the package and then remove the peer dependency, if defined.
+						if (dep.newVersion) {
+							// This approach has the added advantage that scoped overrides should now work, too.
+							pkg.dependencies[dep.peerDependency] = dep.newVersion;
+						}
+						delete pkg.peerDependencies[dep.peerDependency];
+
+						console.log(
+							`      - Overrode ${pkg.name}@${pkg.version} peerDependency ${dep.peerDependency}@${pkg.dependencies[dep.peerDependency]}`,
+						);
+					} catch (err) {
+						console.error(err);
+					}
```

If we run into an error here, I think some will pay attention to the output and realize it, but others might not. To me it feels like an issue here would be better treated as fatal, let it propagate and terminate the process.

---

I don't know what is possibly going to throw in here. I'll just remove the try-catch.